### PR TITLE
Add missing `HTTPRequest` Result descriptions.

### PR DESCRIPTION
**Pull request by syntaxerror247**

### CODE REVIEW

File: doc/classes/HTTPRequest.xml [hunk 1]
```diff
@@ -275,6 +275,7 @@
 			Request successful.
 		</constant>
 		<constant name="RESULT_CHUNKED_BODY_SIZE_MISMATCH" value="1" enum="Result">
+			Request failed due to a mismatch between the expected and actual chunked body size during transfer.
```
Comment (AThousandShips) [inline doc/classes/HTTPRequest.xml:278]: This description needs some more details I think, right now it is just a different way to say what the name of the error says

Comment (syntaxerror247) [reply doc/classes/HTTPRequest.xml]: Ok, now I have added some possible causes too.

**Comment by AThousandShips** [comment]
Style looks good to me but needs someone knowing these parts of the engine to approve first

File: doc/classes/HTTPRequest.xml [hunk 2]
```diff
@@ -295,6 +296,7 @@
 			Request exceeded its maximum size limit, see [member body_size_limit].
 		</constant>
 		<constant name="RESULT_BODY_DECOMPRESS_FAILED" value="8" enum="Result">
+			Request failed due to an error while decompressing the response body. Possible causes include unsupported or incorrect compression format, corrupted data, or incomplete transfer.
```
Comment (Loff3) [inline doc/classes/HTTPRequest.xml:299]:
```suggestion
			The request failed due to an error while decompressing the response body. Possible causes include unsupported or incorrect compression format, corrupted data, or incomplete transfer.
```

File: doc/classes/HTTPRequest.xml [hunk 3]
```diff
@@ -275,6 +275,7 @@
 			Request successful.
 		</constant>
 		<constant name="RESULT_CHUNKED_BODY_SIZE_MISMATCH" value="1" enum="Result">
+			Request failed due to a mismatch between the expected and actual chunked body size during transfer. Possible causes include network errors, server misconfiguration, or issues with chunked encoding.
```
[the remark this answers is not in the document]
Comment (syntaxerror247) [reply doc/classes/HTTPRequest.xml:278]: Thank you for the suggestion! However, I believe 'Request failed...' fits better with the style used throughout the documentation and helps maintain consistency.

File: doc/classes/HTTPRequest.xml (hunk 2 above)
Comment (syntaxerror247) [reply doc/classes/HTTPRequest.xml:299]: Thank you for the suggestion! However, I believe 'Request failed...' fits better with the style used throughout the documentation and helps maintain consistency.

Comment (Loff3) [reply doc/classes/HTTPRequest.xml:299]: Oh yeah makes sense

**Review by mhilbrunner — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**